soc: stm32: Extra tree soc configuration is blocked by /dts/arm/st/mem.h

After configuring an extra tree SOC based on documentation from #9776 "Move soc code under ZEPHYR_PROJECT/soc" , the build stopps with:
```
.../mem.h:128:2: error: #error "Flash, RAM, and CCM sizes not defined for this chip"
 #error "Flash, RAM, and CCM sizes not defined for this chip"
  ^~~~~
```

--- a/dts/arm/st/mem.h
+++ b/dts/arm/st/mem.h
@@ -125,7 +125,9 @@
 #define DT_FLASH_SIZE		__SIZE_K(256)
 #define DT_SRAM_SIZE		__SIZE_K(64)
 #else
-#error "Flash, RAM, and CCM sizes not defined for this chip"
+#if !defined(DT_FLASH_SIZE) || !defined(DT_SRAM_SIZE)
+#error "Flash, RAM, or CCM sizes not defined for this chip"
+#endif /* !DT_FLASH_SIZE || !DT_FLASH_SIZE */
 #endif
 
 #endif /* __DT_BINDING_ST_MEM_H */

